### PR TITLE
Add Element Template Tab

### PR DESCRIPTION
**Pull request by philippfromme**
__Which issue does this PR address?__

Related to https://github.com/camunda/camunda-modeler/issues/1894
Related to https://github.com/camunda/camunda-modeler/issues/1889

__Acceptance Criteria__

* [x] Corresponds to the concept 
* [x] Corresponds to the design 

__Definition of Done__

* [x] corresponds to [the design principles](https://github.com/bpmn-io/design-principles)
* [x] corresponds to [the code standards](https://github.com/bpmn-io/bpmn-js-properties-panel/blob/master/.github/CONTRIBUTING.md#creating-a-pull-request)
* [x] passes Continuous Integration checks
* [x] available as feature branch on GitHub
  * [x] contains the cleaned up commit history
  * [x] commit messages satisfy our [commit message guidelines](https://www.conventionalcommits.org/)
  * [x] a single commit closes the issue via Closes #issuenr

### CODE REVIEW

File: lib/provider/camunda/CamundaPropertiesProvider.js [hunk 1]
```diff
@@ -448,9 +471,16 @@ function createExtensionElementsGroups(element, bpmnFactory, elementRegistry, tr
...
-
+    eventBus,
+    canvas,
+    bpmnFactory,
+    elementRegistry,
+    elementTemplates,
+    modeling,
+    replace,
+    selection,
+    translate
```
Comment (pinussilvestrus) [inline lib/provider/camunda/CamundaPropertiesProvider.js:482]: Should we indicate this as breaking change? Because updates are needed when this provider is used.

File: lib/provider/camunda/element-templates/CustomElementsPropertiesActivator.js [hunk 2]
```diff
@@ -54,48 +43,23 @@ module.exports = CustomElementsPropertiesActivator;
...
-
-  if (entryId === 'elementTemplate-chooser' || isCustomEntry(entry)) {
-    return true;
-  }
-
-  var entriesVisible = template.entriesVisible || DEFAULT_ENTRIES_VISIBLE;
-
-  if (typeof entriesVisible === 'boolean') {
-    return entriesVisible;
+function isEntryVisible(element, entry, group, tab, template) {
```
Comment (pinussilvestrus) [inline lib/provider/camunda/element-templates/CustomElementsPropertiesActivator.js:54]: Lot's of unused arguments there. Are they still needed to prevent breaking changes?

File: styles/properties.less [hunk 3]
```diff
@@ -100,7 +102,7 @@
     }
   }
 
-  button {
+  .action-button {
```
Comment (pinussilvestrus) [inline styles/properties.less:105]: Should we also indicate this as breaking change?

File: lib/provider/camunda/CamundaPropertiesProvider.js [hunk 4]
```diff
@@ -140,6 +143,46 @@ var getListenerLabel = function(param, translate) {
...
+  }
+}
+
+function getProcessOptions(element) {
+  if (is(element, 'bpmn:Participant')) {
+    return { processIdDescription: PROCESS_KEY_HINT };
+  }
+}
+
+function createElementTemplateGroups(element, bpmnFactory, canvas, elementTemplates, modeling, replace, selection, translate) {
```
Comment (pinussilvestrus) [inline lib/provider/camunda/CamundaPropertiesProvider.js:162]: I like a lot that we now separate the template groups from the "normal" state. 👏

File: styles/properties.less (hunk 3 above)
Comment (barmac) [reply styles/properties.less:105]: I believe this is a breaking change for anyone using custom entries. I will have to change my code in order to make buttons look OK again.

File: lib/provider/camunda/CamundaPropertiesProvider.js (hunk 1 above)
Comment (philippfromme) [reply lib/provider/camunda/CamundaPropertiesProvider.js:482]: Yeah, we should probably do this. To avoid breaking your custom provider that extends this one you can do:

```javascript
super(...CamundaPropertiesProvider.$inject.map(dependency => injector.get(dependency)));
```

File: lib/provider/camunda/element-templates/CustomElementsPropertiesActivator.js (hunk 2 above)
Comment (philippfromme) [reply lib/provider/camunda/element-templates/CustomElementsPropertiesActivator.js:54]: I've tried to streamline the events a bit. In this case, I only need the tab but someone might need the entry, the group or the tab to determine if an entry should be visible. I'll remove the unused parameters from this function but they will stay in the method that calls it.

Comment (philippfromme) [reply lib/provider/camunda/element-templates/CustomElementsPropertiesActivator.js]: Breaking changes will be introduced by firing both `propertiesPanel.isEntryVisible` and `propertiesPanel.isEntryEditable` with more information.

File: lib/provider/camunda/CamundaPropertiesProvider.js (hunk 1 above)
Comment (philippfromme) [reply lib/provider/camunda/CamundaPropertiesProvider.js]: I've added a note about the breaking change to the commit.

File: styles/properties.less (hunk 3 above)
Comment (philippfromme) [reply styles/properties.less:105]: Unfortunately, it was necessary. `button { position: absolute; }` is just unacceptable. I will add a note about the breaking change to the respective commit.

File: styles/properties.less [hunk 5]
```diff
@@ -60,6 +60,8 @@
   }
 
   [contenteditable] {
+    min-height: 24px;
```
Comment (pinussilvestrus) [inline styles/properties.less:63]: Why is this change needed?

Comment (philippfromme) [reply styles/properties.less:63]: Just a quick fix for this issue:

Comment (pinussilvestrus) [reply styles/properties.less:63]: Alright, nice finding 👍

**Review by pinussilvestrus — APPROVED** [review]
Changes look good from my perspective 👏 

Two things to consider:
* Will there be an additional Design Review for this PR or will we do it once we merged all things together? @andreasgeier
* There are unmerged changes in the Modeler (https://github.com/camunda/camunda-modeler/commits/1894-element-template-tab) so this will work properly. Will you open another PR afterward? Just to make sure, we don't get those changes lost somehow 😄

**Comment by philippfromme** [comment]
@pinussilvestrus The Camunda Modeler branch will be merged once it includes these changes.
